### ISSUE TITLE
Enhance Configuration, Documentation, and Script Functionality

### ISSUE THREAD

**Issue by Kappuccino111** [jonas-rem/lwm2m_server#20]
- [ ] Add a single file for configuring all requirements.
- [ ] Update documentation.rst to be more comprehensive.
- [x] Resolve issue in zephyr_build_run_sim.sh
- [x] Add a contributing guide

**Comment by jonas-rem** [comment]
The issues in `zephyr_build_run_sim.sh` could be related to setting up the `zeth` network interface via `net-setup`. If the setup worked you should see the `zeth` if with `ip a`

```
17: zeth: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN group default qlen 1000
    link/ether 00:00:5e:00:53:ff brd ff:ff:ff:ff:ff:ff
    inet 192.0.2.2/24 scope global zeth
       valid_lft forever preferred_lft forever
    inet6 2001:db8::2/128 scope global nodad 
       valid_lft forever preferred_lft forever
    inet6 fe80::200:5eff:fe00:53ff/64 scope link proto kernel_ll 
       valid_lft forever preferred_lft forever
```

**Comment by Kappuccino111** [comment]
Closed #22 and #23. Will create a single PR with all the changes, which when merged will automatically accomplish what the individual PRs were doing 👍
